src/sentry/tasks/integrations.py

```diff
@@ -41,9 +33,29 @@ def post_comment(external_issue_id, data, user_id, **kwargs):
     installation = integration.get_installation(
         organization_id=external_issue.organization_id,
     )
+
+    if action != 'delete':
+        try:
+            note = Activity.objects.get(
+                type=Activity.NOTE,
+                id=group_note_id,
+            )
+        except Activity.DoesNotExist:
+            return  # TODO(lb)..... hmmm
+
     if installation.should_sync('comment'):
-        installation.create_comment(external_issue.key, user_id, data['text'])
+
+        if action == 'create':
+            comment = installation.create_comment(external_issue.key, user_id, note.data['text'])
+            note.data['external_id'] = installation.get_comment_id(comment)
+            note.save()
+        elif action == 'delete':
+            installation.delete_comment(external_issue.key, user_id, note.id)
+        elif action == 'update':
+            installation.update_comment(external_issue.key, user_id, note.id, note.data['text'])
```

Comment:
Yes, but unfortunately without having an external id it'll be impossible to know which comment to update. I'm assuming here that this feature will only work on comments that were created after this change was implemented.

Reply:
I worry that sync will be 'sometimes broken' in ways that are hard for the customer/support to figure out.

Reply:
The comment will have a datetime field on the saved model entry. I think that if we give customer support a date for when "all comments made after this date sync" would be ok. Otherwise, we'd need to have some way to figure out which sentry comment corresponded to the comment on Jira. Basically backfilling the data. If you feel that it is important enough to do so, I can see how feasible that would be. I'm not sure though.